ExtraEnemyCustomization/Customizations/Models/Managers/SilhouetteManager.cs

```diff
@@ -0,0 +1,172 @@
+        {
+
+        }
+
+        public void EnableSilhouette()
+        {
+            SetColorB(_LatestColorB);
+        }
+        public void DisableSilhouette()
+        {
+            SetColorB(Color.clear);
+        }
+
+        public void SetColorA(Color color)
+        {
+            SilhouetteMaterial.SetVector("_ColorA", color);
+        }
+
+        public void SetColorB(Color color)
+        {
+            SilhouetteMaterial.SetVector("_ColorB", color);
+            _LatestColorB = color;
+        }
+    }
+
+    public class SilhouetteManager : MonoBehaviour
+    {
+        public EnemyAgent _Agent;
+        public Color _DefaultColor;
+        public bool _RequireTag;
+        public bool _ReplaceColorWithMarker;
+        public bool _KeepOnDead;
+        public float _DeadEffectDelay;
+
+        private bool tagUpdateDone = true;
+        private NavMarker enemyMarker = null;
+        private EnemySilhouette[] silhouettes = null;
+
+        public SilhouetteManager(IntPtr ptr) : base(ptr)
+        {
+
+        }
+
+        internal void Start()
+        {
+            silhouettes = GetComponentsInChildren<EnemySilhouette>(true);
+            foreach(var sil in silhouettes)
+            {
+                sil.SilhouetteMaterial = sil.GetComponent<SkinnedMeshRenderer>().material;
+            }
+
+            if (_RequireTag || _ReplaceColorWithMarker)
+            {
+                EnemyMarkerEvents.RegisterOnMarked(_Agent, OnMarked);
+            }
+            
+            if (!_RequireTag)
+            {
+                SetColorB(_DefaultColor);
+                Show();
+            }
+        }
+
+        internal void Update()
+        {
+            if (!_Agent.Alive)
+            {
+                if (!_KeepOnDead)
+                {
+                    KillSilhouette(_DeadEffectDelay);
+                    enabled = false;
+                }
+            }
+
+            if (!tagUpdateDone)
+            {
+                if (!_Agent.IsTagged)
+                {
+                    if (_RequireTag)
+                    {
+                        Hide();
+                    }
+
+                    tagUpdateDone = true;
+                }
+
+                var alpha = enemyMarker?.m_enemySubObj?.m_sprites?[0]?.color.a ?? 0.0f;
+                var tagColor = enemyMarker?.m_enemySubObj?.m_sprites?[0]?.color ?? _DefaultColor;
+                if (_RequireTag)
+                {
+                    var newColor = _ReplaceColorWithMarker ? tagColor.AlphaMultiplied(alpha) : _DefaultColor.AlphaMultiplied(alpha);
+                    SetColorB(newColor);
+                }
+                else
+                {
+                    var newColor = Color.Lerp(_DefaultColor, tagColor, alpha);
+                    SetColorB(newColor);
+                }
+            }
+        }
+
+        public void OnMarked(EnemyAgent agent, NavMarker marker)
+        {
+            Logger.Log("Marked! " + agent.name);
```

Comment:
I haven't tested this, but wouldn't this print `Marked! [name]` to the console for every marked enemy?